test/validation/api/ipsec/ipsec.c

```diff
@@ -533,17 +535,33 @@ static int recv_pkts_inline(const ipsec_test_part *part,
 
 		ev = odp_queue_deq(suite_context.queue);
 		if (ODP_EVENT_INVALID != ev) {
+			odp_packet_t pkt;
+			int num_pkts = 0;
+
 			CU_ASSERT(odp_event_is_valid(ev) == 1);
-			CU_ASSERT_EQUAL(ODP_EVENT_PACKET,
-					odp_event_types(ev, &subtype));
-			CU_ASSERT_EQUAL(ODP_EVENT_PACKET_IPSEC,
-					subtype);
+			CU_ASSERT_EQUAL(ODP_EVENT_PACKET, odp_event_type(ev));
+			pkt = odp_packet_from_event(ev);
+
 			CU_ASSERT(!part->out[i].status.error.sa_lookup);
 
-			pkto[i] = odp_ipsec_packet_from_event(ev);
-			CU_ASSERT(odp_packet_subtype(pkto[i]) ==
-				  ODP_EVENT_PACKET_IPSEC);
-			i++;
+			if (suite_context.reass_ipv4 || suite_context.reass_ipv6) {
```

Comment:
I do not think this check is needed here. The code could always check odp_packet_reass_status() and act accordingly. In addition to simplifying the code, it would have the benefit that non-reassembly tests would always perform the same actions, regardless of whether reassembly is supported or not.

Reply:
Agreed. Will address in next version.